packages/dataviews/src/types.ts

```diff
@@ -418,6 +418,11 @@ interface ViewBase {
 	 * Whether to show the hierarchical levels.
 	 */
 	showLevels?: boolean;
+
+	/**
+	 * The field to group by.
+	 */
+	groupByField?: string;
```

Comment:
Any thoughts on a UI to configure this? (not necessary for this PR)

Also would be good to document this in the README.

Reply:
Done at https://github.com/WordPress/gutenberg/pull/70752/commits/d97929ebd38a4270e0ea2ea4c00d32ad67777d66

Reply:
> Any thoughts on a UI to configure this? (not necessary for this PR)

The obvious way is to add a new select in the view config somewhere. Alternatively, we could explore making it part of the field list (similarly to what we have for the media field).